crates/circuit/src/dag_circuit.rs

```diff
@@ -6315,6 +6315,60 @@ impl DAGCircuit {
         }
     }
 
+    /// Insert an op given by callback on each individual qubit into a node
+
+    #[allow(unused_variables)]
+    pub fn replace_on_incoming_qubits<F>(
+        &mut self,
+        py: Python, // Unused if cache_pygates isn't enabled
+        node: NodeIndex,
+        mut insert: F,
```

Comment:
Does `F` need to be mutable?

Reply:
It does because I used `FnMut` for the the type here. It doesn't have to be mutable if we accepted an `Fn` for `F` instead. I just wasn't sure which kind of callables we'd allow longer term and opted to be a bit more conservative in the typing, and having `FnMut` means the callable is allowed to mutate it's environment. I can change it to `Fn` in this case because the only callback we have so far doesn't need to be `FnMut`.

Reply:
In this case, `FnMut` is probably better as it allows the user more flexibility, it especially allows users to easily track the changes performed by `replace_on_incoming_qubits`.

Reply:
Changed in: https://github.com/Qiskit/qiskit/pull/13025/commits/2ff94f5308bfc0b61391ce45b4fd8d0dcdbfde26